napari/resources/build_icons.py

```diff
@@ -171,14 +171,16 @@ def build_pyqt_resources(out_path: str, overwrite: bool = False) -> str:
     # once, we use this try/catch loop because
     # 1. we always want to use pyrcc5 if it's available, regardless of API
     # 2. it will sometimes, (if not always) be named pyrcc5.bat on windows...
-    # 3. only then do we try pyside2-rcc
-    # see https://github.com/napari/napari/issues/1221 for background
-    for binary in ('pyrcc5.bat', 'pyrcc5', 'pyside2-rcc'):
-        try:
-            check_call([binary, '-o', out_path, qrc_path])
-        except FileNotFoundError:
-            continue
-        break
+    #    but shutil.which() will find that too
+    # 3. We also want to prefer binaries higher up on the path
+    # 4. only then do we try pyside2-rcc
+    # see https://github.com/napari/napari/issues/1221
+    # and https://github.com/napari/napari/issues/1254
+    for bin_name in ('pyrcc5', 'pyside2-rcc'):
+        rcc_binary = shutil.which(bin_name)
+        if rcc_binary:
+            check_call([rcc_binary, '-o', out_path, qrc_path])
+            break
```

Comment:
@tlambert03 This still seems risky. It's possible to call napari with `/path/to/conda/env/bin/python -m napari`, in which case the `shutil.which` will still, I believe, find `pyrcc` from a different environment. Why not check for pyrcc in the same environment as `sys.executable`?

Reply:
good point, hadn't thought of that.
but when you say the "same environment as `sys.executable` what exactly do you mean by "environment"?  do you mean "directory"?  or do you mean "figure out what environment that python executable is in"?  If the former, I guess maybe I'm (perhaps unreasonably) afraid of restricting it to a single directory like that.   Also, I think where these things live is perhaps platform specific, on windows they are not in the same directory: 

```shell
(napdev) C:\Users\tjl10\python\napari> where pyside2-rcc
C:\Users\tjl10\miniconda3\envs\napdev\Scripts\pyside2-rcc.exe

(napdev) C:\Users\tjl10\python\napari> where python
C:\Users\tjl10\miniconda3\envs\napdev\python.exe
```

if you mean the latter... then how do we figure out what environment that python executable is "supposed" to be in without making major assumptions about the environment manager and platform?

One possibility here is to keep using `shutil.which` but provide it with a `path` argument, based on `os.getenv('PATH')` ... but with various additional folders (like `sys.executable`, and maybe also that `Scripts` folder for windows) prepended?

Reply:
Wow, that's highly different from *nix envs! Yikes.

```
jni@jni-x1c6 Thu May 21 08:59 
 ~ 
 $ which pyside2-rcc
/home/jni/miniconda3/envs/all/bin/pyside2-rcc

jni@jni-x1c6 Thu May 21 08:59 
 ~ 
 $ which python
/home/jni/miniconda3/envs/all/bin/python
```

And of course there's no accounting for non-conda envs/Pythons...

My suggestion so far is:

- yes, as you say, try shutil.which(path=os.path.dirname(sys.executable)) on *nix, shutil.which(os.path.join(os.path.dirname, r'\Scripts')) on Windows
- if that fails, try your current strategy

That should cover a fair few use cases I think...

Reply:
> Wow, that's highly different from *nix envs! Yikes.

yeah... why can't they all just agree to do the same thing?? 😤

> suggestion so far...

I was thinking combine all that into a single step:

```python

shutil.which(path=os.path.dirname(sys.executable))
os.path.dirname(sys.executable)

Reply:
oh I was totally unaware that you could pass *multiple* paths. In that case, go for it, that's awesome! And yes you should add the Scripts dir on windows!

Reply:
`os.getenv("PATH")` probably needs splitting on `':'`?

Reply:
actually, no, on `os.pathsep`. :grimacing:

Reply:
yeah, I'll make sure to look at the actual values on all platforms when I test.